src/utilities.cpp

```diff
@@ -0,0 +1,25 @@
+#include "utilities.hpp"
+
+std::string trimTrailingWhiteSpaces(const std::string &str)
+{
+    std::string s(str);
+
+    s.erase(
+        find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(isspace))).base(),
+        s.end()
+    );
+    return s;
+}
+
+std::vector<std::string>	split(const std::string &str, char delimiter)
+{
+    std::vector<std::string>	tokens;
+    std::stringstream			ss(str);
+    std::string					token;
+
+    while (std::getline(ss >> std::ws, token, delimiter)) {
+		token = trimTrailingWhiteSpaces(token);
+        tokens.push_back(token);
+    }
+    return tokens;
```

Comment:
I would advise against stringstream (even though this function won't be called that often. Also a way of passing a multi char delim would be nice. 
Example from [SA](https://stackoverflow.com/a/46931770):
```
std::vector<std::string> split(std::string s, std::string delimiter) {
    size_t pos_start = 0, pos_end, delim_len = delimiter.length();
    std::string token;
    std::vector<std::string> res;

    while ((pos_end = s.find(delimiter, pos_start)) != std::string::npos) {
        token = s.substr (pos_start, pos_end - pos_start);
        pos_start = pos_end + delim_len;
        res.push_back (token);
    }

    res.push_back (s.substr (pos_start));
    return res;
}

Reply:
It's another way of doing it, sure why not.